project/data_preparation/dataloader_service.py

```diff
@@ -0,0 +1,65 @@
+import torchvision.transforms as TF
+import torchvision.datasets as datasets
+from torchvision.datasets import ImageFolder
+from torch.utils.data import DataLoader, random_split
+
+class DeviceDataLoader():
+    def __init__(self, dataloader, device):
+        self.dataloader = dataloader
+        self.device = device
+
+    def __iter__(self):
+        for b in self.dataloader:
+            yield [item.to(self.device) for item in b]
+
+    def __len__(self):
+        return len(self.dataloader)
+
+
+def get_dataset(dataset_name='Flowers'):
+    transforms = TF.Compose(
+        [
+            TF.ToTensor(),
+            TF.Resize((32, 32), 
+                      interpolation=TF.InterpolationMode.BICUBIC, 
+                      antialias=True),
+            TF.RandomHorizontalFlip(),
+            TF.Lambda(lambda t: (t * 2) - 1) # Scale between [-1, 1] 
+        ]
+    )
+    
+    if dataset_name == "Flowers":
+        dataset = ImageFolder(root="flowers-recognition/flowers", transform=transforms)
+    elif dataset_name == "CelebA":
+        dataset = ImageFolder(root="celeba-dataset/img_align_celeba", transform=transforms)
+    else:
+        raise ValueError(f"Dataset {dataset_name} is not supported.")
+    
+    return dataset
+
+def split_dataset(dataset, train_ratio=0.7, val_ratio=0.15, test_ratio=0.15):
```

Comment:
```suggestion
def split_dataset(dataset, train_ratio=0.7, val_ratio=0.15):
```
You don't use test_ratio, as that is always the remainder.